\subsection{Genomes and protein-coding regions used}
We used publicly available nucleotide sequences corresponding to 61 genomes of
organisms from major domains of life (Supplement 1). Since these genomes have
varying annotation quality, we adhered to a heuristic approach and analyzed
each genome for open reading frames of minimum length 150 b.p. as proxies to
encoded putative protein sequences (PPS). These regions may or may not
correspond to actual genes, because splicing was not taken into account. We did
not use any of the existing annotations so that well-annotated and poorly
annotated genomes were handled in the same way, and no bias corresponding to
the different annotation processes used by different authors could have
influenced our results. That way we also did our best not to miss any of the
non-annotated protein-coding sequences. As a result, for each genome considered
we created a set of PPSs.
